k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeUtility.java

```diff
@@ -32,6 +34,8 @@
     private static final String HEADER_PARAM_FORMAT = "format";
     private static final String HEADER_FORMAT_FLOWED = "flowed";
 
+    private static int filledBytes = 0;
```

Comment:
I thought of another approach, it is as follows : 
1. Create a `ProgressCallback` interface with one method `onUpdate()`.
2. Create a copy of `createBody()` function that takes an object of `ProgressCallback` as an extra parameter and calls `onUpdate()` every 50 millisecond while the input stream is being copied.
3. In the implementation of the `onUpdate` method we can update the progress in the dialog.

Is this approach correct ?